File: packages/python/src/cachetta/utils/get_extension.py
from pathlib import Path

from ..exceptions import InvalidPathError


def get_extension(cache_path: str | Path) -> str:
    cache_path = str(cache_path)
    parts = cache_path.split('/').pop().split('.')
    if len(parts) == 1:
        raise InvalidPathError('Missing extension for path: %s' % cache_path)
    ext = parts[-1]
    return ext

File: packages/javascript/src/utils/get-extension.ts
import { PathLike } from 'fs';
import { CachettaError } from '../errors.js';

export function getExtension(cachePath: string | PathLike): string {
  const pathStr = String(cachePath);
  const parts = pathStr.split('/').pop()?.split('.');

  if (!parts || parts.length === 1 || parts[parts.length - 1] === '') {
    throw new CachettaError(`Missing file extension: ${cachePath}`);
  }

  return parts[parts.length - 1];
}
